profile/src/main/java/com/scottlogic/deg/profile/reader/ConstraintReader.java

```diff
@@ -207,58 +175,43 @@ private Constraint readGrammaticalConstraintDto(ConstraintDTO dto, ProfileFields
         }
     }
 
-    private int integer(Object value)
+    @Nullable
+    private Object readAnyType(Field field, Object value)
     {
-        return NumberUtils.coerceToBigDecimal(value).intValueExact();
+        switch (field.getType())
+        {
+            case DATETIME:
+                return HelixDateTime.create((String) value).getValue();
+            case NUMERIC:
+                return NumberUtils.coerceToBigDecimal(value);
```

Comment:
If we use HelixDateTime.create for DateTime, should HelixNumber.create handle this case?

Reply:
Ideally HelixDateTime will be present throughout and will not just be used for validation only